### ISSUE TITLE
Interpreter crash on cross in fold

### ISSUE THREAD

**Issue by iFreilicht** [uiua-lang/uiua#118]
```
x ← (
  ≡'□⇡△.
  ⊃⊢'↘1
  ∧(⊠⊂⊔)
)

x [0_1_0 1_0_1 1_0_0 0_1_0]
```

Backtrace:

```
The application panicked (crashed).
Message:  assertion `left == right` failed: shape [4, 3, 2] does not match data length 8
  left: 24
 right: 8
Location: /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/core/src/ops/function.rs:250

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
Run with RUST_BACKTRACE=full to include source snippets.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
                              ⋮ 13 frames hidden ⋮                              
14: core::panicking::assert_failed_inner::hb7d1dba8d154f274
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/core/src/panicking.rs:<unknown line>
15: core::panicking::assert_failed::h05360b8844a39b8a
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/core/src/panicking.rs:280
16: uiua::array::validate_shape::h66d85b2b28cbcef3
    at /Users/feuh/repos/uiua/src/array.rs:73
17: uiua::array::Array<T>::validate_shape::hdfa89e8b0654b7ea
    at /Users/feuh/repos/uiua/src/array.rs:93
18: core::ops::function::FnOnce::call_once::h6c75ff47dbf75084
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/core/src/ops/function.rs:250
19: uiua::value::Value::generic_ref::hddc59b7fcc24b2bb
    at /Users/feuh/repos/uiua/src/value.rs:253
20: uiua::value::Value::validate_shape::h32e079f8ca77943b
    at /Users/feuh/repos/uiua/src/value.rs:206
21: uiua::algorithm::loops::cross::h3eaf4b6d40597877
    at /Users/feuh/repos/uiua/src/algorithm/loops.rs:898
22: uiua::primitive::<impl uiua::primitive::defs::Primitive>::run::h6c7c6f186e8ff200
    at /Users/feuh/repos/uiua/src/primitive/mod.rs:400
23: uiua::run::Uiua::exec::{{closure}}::hd9172a573ec5115e
    at /Users/feuh/repos/uiua/src/run.rs:405
24: uiua::run::Uiua::exec::h664649be7aace488
    at /Users/feuh/repos/uiua/src/run.rs:403
25: uiua::run::Uiua::call_function_with_span::hc66ada346fe71d51
    at /Users/feuh/repos/uiua/src/run.rs:525
26: uiua::run::Uiua::call_with_span::hcea632c6d8d94b4a
    at /Users/feuh/repos/uiua/src/run.rs:515
27: uiua::run::Uiua::call::h3bd80f92fd57caed
    at /Users/feuh/repos/uiua/src/run.rs:536
28: uiua::run::Uiua::call_catch_break::h2fd6d9f21114c82d
    at /Users/feuh/repos/uiua/src/run.rs:559
29: uiua::algorithm::loops::generic_fold1::h960958a4df209250
    at /Users/feuh/repos/uiua/src/algorithm/loops.rs:147
30: uiua::algorithm::loops::fold::hb88721e8af1b5148
    at /Users/feuh/repos/uiua/src/algorithm/loops.rs:107
31: uiua::primitive::<impl uiua::primitive::defs::Primitive>::run::h6c7c6f186e8ff200
    at /Users/feuh/repos/uiua/src/primitive/mod.rs:394
32: uiua::run::Uiua::exec::{{closure}}::hd9172a573ec5115e
    at /Users/feuh/repos/uiua/src/run.rs:405
33: uiua::run::Uiua::exec::h664649be7aace488
    at /Users/feuh/repos/uiua/src/run.rs:403
34: uiua::run::Uiua::call_function_with_span::hc66ada346fe71d51
    at /Users/feuh/repos/uiua/src/run.rs:525
35: uiua::run::Uiua::call_with_span::hcea632c6d8d94b4a
    at /Users/feuh/repos/uiua/src/run.rs:515
36: uiua::run::Uiua::exec::{{closure}}::h0f193b16588bf630
    at /Users/feuh/repos/uiua/src/run.rs:411
37: core::result::Result<T,E>::and_then::h16eae5d1bb1daab7
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/core/src/result.rs:1320
38: uiua::run::Uiua::exec::h664649be7aace488
    at /Users/feuh/repos/uiua/src/run.rs:409
39: uiua::run::Uiua::exec_global_instrs::h24b9db68d39b71e7
    at /Users/feuh/repos/uiua/src/run.rs:343
40: uiua::compile::<impl uiua::run::Uiua>::item::h3a2193328558b9b8
    at /Users/feuh/repos/uiua/src/compile.rs:71
41: uiua::compile::<impl uiua::run::Uiua>::items::h65367a0af847d644
    at /Users/feuh/repos/uiua/src/compile.rs:25
42: uiua::run::Uiua::load_impl::{{closure}}::h82705b3cbe053834
    at /Users/feuh/repos/uiua/src/run.rs:283
43: <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::ha5fbce0ed1460d55
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/core/src/panic/unwind_safe.rs:271
44: std::panicking::try::do_call::hdaa7d5300a2680b7
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/std/src/panicking.rs:504
45: ___rust_try
    at <unknown source file>
46: std::panicking::try::h05d9f923d1c06bbf
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/std/src/panicking.rs:468
47: std::panic::catch_unwind::h9b294e7b866b615e
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/std/src/panic.rs:142
48: uiua::run::Uiua::load_impl::h9a13a48612408e09
    at /Users/feuh/repos/uiua/src/run.rs:283
49: uiua::run::Uiua::load_file::h6f2a6d55437ce206
    at /Users/feuh/repos/uiua/src/run.rs:238
50: uiua::run::h1134826e34f250c1
    at /Users/feuh/repos/uiua/src/main.rs:121
51: uiua::main::h1d9de589ddeb04b9
    at /Users/feuh/repos/uiua/src/main.rs:44
52: core::ops::function::FnOnce::call_once::h58dcd301614282e8
    at /rustc/960754090acc9cdd2a5a57586f244c0fc712d26c/library/core/src/ops/function.rs:250
                              ⋮ 13 frames hidden ⋮                              
Error: The interpreter has crashed!
Hooray! You found a bug!
Please report this at http://github.com/uiua-lang/uiua/issues/new

code:
x.ua:4:5
x ← (
  ≡'□⇡△.
  ⊃⊢'↘1
  ∧(⊠⊂⊔)
)

x [0_1_0 1_0_1 1_0_0 0_1_0]

   ╭─[x.ua:4:5]
 4 │   ∧(⊠⊂⊔)
   │     ─  
───╯
```

**Comment by kaikalii** [comment]
Should be fixed in cb7af8c70c7139225

**Comment by iFreilicht** [comment]
Jup, tested it on current master :+1:
